app/src/lib/shells/win32.ts

```diff
@@ -215,6 +228,44 @@ async function findGitBash(): Promise<string | null> {
   return null
 }
 
+async function findCygwin(): Promise<string | null> {
+  const registryPath64 = enumerateValues(
+    HKEY.HKEY_LOCAL_MACHINE,
+    'SOFTWARE\\Cygwin\\setup'
+  )
+  const registryPath32 = enumerateValues(
+    HKEY.HKEY_LOCAL_MACHINE,
+    'SOFTWARE\\WOW6432Node\\Cygwin\\setup'
+  )
+
+  if (registryPath64.length === 0 && registryPath32.length === 0) {
+    return null
+  }
+
+  let registryPath
```

Comment:
:art: this could be rewritten as a oneliner:

```ts
  // always favour 64-bit registry path, if found
  const registryPath =
    registryPath64.length > 0 ? registryPath64 : registryPath32
```